### PR TITLE
Force dependency plugin(s) to load first - Fix #253

### LINKED ISSUE #253: Controlling the loading order of plugins
OK, Operating system change issue number 2...

  On WIndows, wrapper loaded the plugins alphabetically.  Therefore plugin "global.py" loaded prior to "stafftools.py" and "simple.py".. now it is loading AFTER them... and since they use the getplugincontext (depend on global.py), they error out during the plugin loading:
```
[18:25:10] [Wrapper.py/INFO] Loading plugins...
[18:25:10] [Wrapper.py/INFO] Loading plugin stafftools.py...
[18:25:10] [Wrapper.py/DEBUG] Traceback (most recent call last):
[18:25:10] [Wrapper.py/DEBUG]   File "/home/surest/vanilla/wrapper/plugins.py", line 93, in loadPlugins
[18:25:10] [Wrapper.py/DEBUG]     elif i[-3:] == ".py": self.loadPlugin(i)
[18:25:10] [Wrapper.py/DEBUG]   File "/home/surest/vanilla/wrapper/plugins.py", line 56, in loadPlugin
[18:25:10] [Wrapper.py/DEBUG]     main = plugin.Main(API(self.wrapper, name, id), PluginLog(self.log, name))
[18:25:10] [Wrapper.py/DEBUG]   File "wrapper-plugins/stafftools.py", line 25, in __init__
[18:25:10] [Wrapper.py/DEBUG]     self.globalset = api.getPluginContext("com.suresttexas00.plugins.global")
[18:25:10] [Wrapper.py/DEBUG]   File "/home/surest/vanilla/wrapper/api/__init__.py", line 136, in getPluginContext
[18:25:10] [Wrapper.py/DEBUG]     raise NonExistentPlugin("Plugin %s does not exist!" % id)
[18:25:10] [Wrapper.py/DEBUG] NonExistentPlugin: Plugin com.suresttexas00.plugins.global does not exist!
[18:25:10] [Wrapper.py/DEBUG] 
[18:25:10] [Wrapper.py/ERROR] Failed to import plugin 'stafftools.py'
[18:25:10] [Wrapper.py/INFO] Loading plugin voterewards.py...
[18:25:10] [Wrapper.py/INFO] Loading plugin broadcaster.py...
[18:25:10] [Wrapper.py/DEBUG] [Broadcaster] Registered event 'timer.second'
[18:25:10] [Wrapper.py/INFO] Loading plugin regions.py...
[18:25:10] [Wrapper.py/DEBUG] [Regions] Registered help group 'Regions' with 14 commands
....
[18:37:49] [Wrapper.py/INFO] Loading plugin simple.py...
[18:37:49] [Wrapper.py/DEBUG] Traceback (most recent call last):
[18:37:49] [Wrapper.py/DEBUG]   File "/home/surest/vanilla/wrapper/plugins.py", line 93, in loadPlugins
[18:37:49] [Wrapper.py/DEBUG]     elif i[-3:] == ".py": self.loadPlugin(i)
[18:37:49] [Wrapper.py/DEBUG]   File "/home/surest/vanilla/wrapper/plugins.py", line 56, in loadPlugin
[18:37:49] [Wrapper.py/DEBUG]     main = plugin.Main(API(self.wrapper, name, id), PluginLog(self.log, name))
[18:37:49] [Wrapper.py/DEBUG]   File "wrapper-plugins/simple.py", line 23, in __init__
[18:37:49] [Wrapper.py/DEBUG]     self.globalset = api.getPluginContext("com.suresttexas00.plugins.global")
[18:37:49] [Wrapper.py/DEBUG]   File "/home/surest/vanilla/wrapper/api/__init__.py", line 136, in getPluginContext
[18:37:49] [Wrapper.py/DEBUG]     raise NonExistentPlugin("Plugin %s does not exist!" % id)
[18:37:49] [Wrapper.py/DEBUG] NonExistentPlugin: Plugin com.suresttexas00.plugins.global does not exist!
[18:37:49] [Wrapper.py/DEBUG] 
[18:37:49] [Wrapper.py/ERROR] Failed to import plugin 'simple.py'
[18:37:49] [Wrapper.py/INFO] Loading plugin global.py...
Global [[18:37:49]/DEBUG] passed up on bad item: MaxPlayersPerIP = 3
Global [[18:37:49]/DEBUG] NewPlayerMessage = &aWelcome to the server %s.  Type /help to begin!
Global [[18:37:49]/DEBUG] SlashSpawnCoords = 3259.5,104,2669.5,0,0
Global [[18:37:49]/DEBUG] toolbox_useVanillaBehavior = false
Global [[18:37:49]/DEBUG] useGlobalChat = true
Global [[18:37:49]/DEBUG] teleport_useVanillaBehavior = true
Global [[18:37:49]/DEBUG] ReturningPlayerMessage = &aWelcome back, %s! Most /help commands are clickable.
Global [[18:37:49]/DEBUG] toolbox_usePermissions = true
Global [[18:37:49]/DEBUG] claims_useVanillaBehavior = false
Global [[18:37:49]/DEBUG] teleport_usePermissions = true
Global [[18:37:49]/DEBUG] defaultOp = SurestTexas00
Global [[18:37:49]/DEBUG] /n
[18:37:49] [Wrapper.py/DEBUG] [Global] Registered command 'global'
[18:37:49] [Wrapper.py/DEBUG] [Global] Registered event 'timer.second'
[18:37:49] [Wrapper.py/DEBUG] [Global] Registered event 'player.login'
[18:37:49] [Wrapper.py/DEBUG] [Global] Registered event 'player.death'
[18:37:49] [Wrapper.py/DEBUG] [Global] Registered event 'player.rawMessage'
[18:37:49] [Wrapper.py/DEBUG] [Global] Registered event 'player.chatbox'
[18:37:49] [Wrapper.py/INFO] Loading plugin portals.py...
...
```

### PR DESCRIPTION
**Pull request by suresttexas00**
Some plugins using the getPluginContext(somepluginID) can error out and not be loaded if the plugin "somepluginID" is not already loaded.

I added some code to allow those plugins to specify a list of plugin(s) that must be loaded first.  Once successfully loaded, further attempts to load that plugin are ignored by the code.  Also made this code and the previous plugin.DISABLED option code more consistent with the rest of the plugin code.

### CODE REVIEW

File: src/plugins.py [hunk 1]
```diff
@@ -35,7 +35,9 @@ def loadPlugin(self, i):
 		else:
 			return False
 		try: disabled = plugin.DISABLED
-		except: plugin.DISABLED = False
```
Comment (benbaptist) [inline src/plugins.py]: Ugh... looking at this code makes me barf. I don't know what toxins I was on writing this. This could use a bit of a re-write.

Comment (suresttexas00) [inline src/plugins.py]: Well, I understand how you feel... On so many levels!  The plugins I first wrote are a __MESS__!  A disgusting revolting vile mess...

File: src/plugins.py [hunk 2]
```diff
@@ -50,9 +52,15 @@ def loadPlugin(self, i):
 		except: author = None
 		try: website = plugin.WEBSITE
 		except: website = None
-		if id in self.wrapper.storage["disabled_plugins"] or plugin.DISABLED:
```
Comment (suresttexas00) [inline src/plugins.py]: and by the way,  this part was my fault (not assigning `plugin.DISABLED` to `disabled` and as a result making this "if" statement test against plugin.Disabled).  So this part was a correction to an earlier PR of mine.

**Comment by suresttexas00** [comment]
So what is the proper way to test for a variables existence in another module without an error?

I definitely agree this code could use a lot less try: excepts:

**Comment by suresttexas00** [comment]
Included with #272

**Pull request closed without merging**